### PR TITLE
#1897 refactoring and completion on Static Var Compensator models

### LINKED ISSUE #1897: Complete SVarC library for DynaFlow
For the moment we have an SVarCPV model that does a V + Lambda * Q control.
If Lambda = 0, which is equivalent to a fully PV model, the simulation fails (because of the way the model is implemented).

I suggest we create a specific model for cases where Lambda = 0, with the same logic as what has been done for GeneratorPVSignalN in DynaFlow in terms of voltage regulation (there isn't the possibility here to use a BVarRawPu variable like in the case where Lambda is not zero).
We could name it SVarCPV and rename the existing one SVarCPVQ or SVarCPVProp

All extensions and lambda<>0 and regulationMode="VOLTAGE"→ SVarCPVQ
All extensions and lambda = 0 and regulationMode="VOLTAGE" → SVarCPV 
No lambda and regulationMode="VOLTAGE" → SVarCPV

No standbyAutomaton and lambda<>0 and regulationMode="VOLTAGE" → SvarCPVQSimplified (no mode change)
No standbyAutomaton and lambda=0 and regulationMode="VOLTAGE" → SvarCPVSimplified (no mode change)
No standbyAutomaton and No lambda and  regulationMode="VOLTAGE" → SvarCPVSimplified (no mode change)

Otherwise → nothing in .dyd

### PR DESCRIPTION
**Pull request by marcochiaramello**
Closes #1897

### CODE REVIEW

File: dynawo/sources/Models/Modelica/PreassembledModels/StaticVarCompensatorPVPropSimplified.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+    Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/PreassembledModels/StaticVarCompensatorPVPropSimplified.xml:3]: 2021 (for other files also)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo [hunk 2]
```diff
@@ -22,50 +22,57 @@ model SVarCPV "PV static var compensator model"
   extends AdditionalIcons.SVarC;
   extends SwitchOff.SwitchOffShunt;
 
+  type BStatus = enumeration (Standard "Susceptance is between its maximal and minimal values",
+                              SusceptanceMax "Susceptance is fixed to its maximal value",
+                              SusceptanceMin "Susceptance is fixed to its minimal value");
+
   Connectors.ACPower terminal(V(re(start = u0Pu.re), im(start = u0Pu.im)), i(re(start = i0Pu.re), im(start = i0Pu.im))) "Connector used to connect the static var compensator to the grid";
 
   extends BaseControls.Parameters.Params_ModeHandling;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo:31]: No import needed ?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo [hunk 3]
```diff
@@ -22,50 +22,57 @@ model SVarCPV "PV static var compensator model"
...
   Types.ActivePowerPu PInjPu(start = 0) "Active power in p.u (base SnRef) (generator convention)";
-  Types.VoltageModulePu URefPu(start = URef0 / UNom) = modeHandling.URefPu "Reference voltage amplitude in p.u (base UNom)";
+  Types.VoltageModulePu URefPu(start = URef0Pu) = modeHandling.URefPu "Reference voltage amplitude in p.u (base UNom)";
 
-  BaseControls.ModeHandling modeHandling(Mode0 = Mode0, UNom = UNom, URefDown = URefDown, URefUp = URefUp, UThresholdDown = UThresholdDown, UThresholdUp = UThresholdUp, tThresholdDown = tThresholdDown, tThresholdUp = tThresholdUp, URef0 = URef0);
+  BaseControls.ModeHandling modeHandling(Mode0 = Mode0, UNom = UNom, URefDown = URefDown, URefUp = URefUp, UThresholdDown = UThresholdDown, UThresholdUp = UThresholdUp, tThresholdDown = tThresholdDown, tThresholdUp = tThresholdUp, URef0 = URef0Pu * UNom);
 
 protected
   parameter Types.PerUnit B0Pu "Start value of the susceptance in p.u (base UNom, SnRef)";
   parameter Types.VoltageModulePu U0Pu  "Start value of voltage amplitude at injector terminal in p.u (base UNom)";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo:53]: One space and not two before comment (sorry, I think it was my mistake) (also for other parameters)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo [hunk 4]
```diff
@@ -22,50 +22,57 @@ model SVarCPV "PV static var compensator model"
...
   parameter Types.ComplexCurrentPu i0Pu  "Start value of complex current at injector terminal in p.u (base UNom, SnRef) (receptor convention)";
-  parameter Types.VoltageModule URef0  "Start value of voltage reference in kV";
   final parameter Types.PerUnit BVar0Pu = B0Pu - BShuntPu "Start value of variable susceptance in p.u (base UNom, SnRef)";
   parameter BaseControls.Mode Mode0 "Start value for mode";
   parameter Boolean selectModeAuto0 = true "Start value of the boolean indicating whether the SVarC is initially in automatic configuration";
   final parameter Integer setModeManual0 = Integer(Mode0) "Start value of the mode when in manual configuration";
+  BStatus bStatus (start = BStatus.Standard) "Susceptance value status: standard, susceptancemax, susceptancemin";
 
 equation
-  UPu = Modelica.ComplexMath.'abs'(terminal.V);
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPV.mo:63]: Please remove blank line after equation

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPVProp.extvar [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+    Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPVProp.extvar:3]: 2021

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPVProp.mo [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  import Dynawo.Types;
+  import Dynawo.Connectors;
+  import Dynawo.Electrical.StaticVarCompensators.BaseControls.Mode;
+  import Dynawo.Electrical.Controls.Basics.SwitchOff;
+
+  extends AdditionalIcons.SVarC;
+  extends SwitchOff.SwitchOffShunt;
+
+  Connectors.ACPower terminal(V(re(start = u0Pu.re), im(start = u0Pu.im)), i(re(start = i0Pu.re), im(start = i0Pu.im))) "Connector used to connect the static var compensator to the grid";
+
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/SVarCPVProp.mo:26]: Why not also having a BStatus for the prop models ?

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/CMakeLists.txt [hunk 7]
```diff
@@ -18,9 +18,20 @@ set(MODEL_FILES
   SVarCPV.mo
   SVarCPV.extvar
   SVarCPV_INIT.mo
+  SVarCPVProp.mo
+  SVarCPVProp.extvar
+  SVarCPVPropSimplified.mo
+  SVarCPVPropSimplified.extvar
   SVarCPVRemote.mo
   SVarCPVRemote.extvar
-  SVarCPVRemote_INIT.mo
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/CMakeLists.txt:23]: yes!

File: .gitignore [hunk 8]
```diff
@@ -13,6 +13,7 @@ distributions/
 nrt/data/*/outputs*/
 nrt/data/*/*/outputs*/
 nrt/data/*/*/*/outputs*/
+nrt/data/*/*/*/*/outputs*/
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply .gitignore:16]: no, references are kept

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/CMakeLists.txt [hunk 9]
```diff
@@ -18,9 +18,20 @@ set(MODEL_FILES
   SVarCPV.mo
   SVarCPV.extvar
   SVarCPV_INIT.mo
+  SVarCPVProp.mo
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/StaticVarCompensators/CMakeLists.txt]: done

**Review by cossartque — APPROVED** [review]

**Pull request merged**